qiskit/transpiler/passes/optimization/optimize_1q_commutation.py

```diff
@@ -106,8 +113,10 @@ def _commute_through(blocker, run, front=True):
 
         if run == []:
             return [], []
-
-        run_clone = copy(run)
+        # use deque to have modification
+        # operations which are constant
+        # time
+        run_clone = deque(copy(run))
```

Comment:
Is the `copy()` call still needed here? I would assume converting to a deque is effectively a shallow copy too as it iterates over the input.